data-prepper-plugins/kafka-plugins/src/main/java/org/opensearch/dataprepper/plugins/kafka/util/KafkaTopicMetrics.java

```diff
@@ -0,0 +1,197 @@
+            }
+        });
+    }
+
+    public void register(final KafkaConsumer consumer) {
+        metricValues.put(consumer, new HashMap<>());
+        final Map<String, Double> consumerMetrics = metricValues.get(consumer);
+        metricsNameMap.forEach((k, name) -> {
+            consumerMetrics.put(k, 0.0);
+        });
+    }
+
+    public Counter getNumberOfRecordsCommitted() {
+        return numberOfRecordsCommitted;
+    }
+
+    public Counter getNumberOfNonConsumers() {
+        return numberOfNonConsumers;
+    }
+
+    public Counter getNumberOfPollAuthErrors() {
+        return numberOfPollAuthErrors;
+    }
+
+    public Counter getNumberOfBufferSizeOverflows() {
+        return numberOfBufferSizeOverflows;
+    }
+
+    public Counter getNumberOfDeserializationErrors() {
+        return numberOfDeserializationErrors;
+    }
+
+    public Counter getNumberOfRecordsFailedToParse() {
+        return numberOfRecordsFailedToParse;
+    }
+
+    public Counter getNumberOfNegativeAcknowledgements() {
+        return numberOfNegativeAcknowledgements;
+    }
+
+    public Counter getNumberOfPositiveAcknowledgements() {
+        return numberOfPositiveAcknowledgements;
+    }
+
+    public String getTopicMetricName(final String metricName) {
+        return "topic."+topicName+"."+metricName;
+    }
+
+    private String getCamelCaseName(final String name) {
+        String camelCaseName = metricsNameMap.get(name);
+        if (Objects.isNull(camelCaseName)) {
+            return name;
+        }
+        return camelCaseName;
+    }
+
+    public void update(final KafkaConsumer consumer) {
+        Map<MetricName, ? extends Metric> metrics = consumer.metrics();
+
+        synchronized(metricValues) {
+            Map<String, Double> consumerMetrics = metricValues.get(consumer);
+            for (Map.Entry<MetricName, ? extends Metric> entry : metrics.entrySet()) {
+                MetricName metric = entry.getKey();
+                Metric value = entry.getValue();
+                String metricName = metric.name();
+                if (Objects.nonNull(metricsNameMap.get(metricName))) {
+                    if (metric.tags().containsKey("partition") &&
+                       (metricName.equals("records-lag-max") || metricName.equals("records-lead-min"))) {
+                       continue;
+                    }
+
+                    if (metricName.contains("consumed-total") && !metric.tags().containsKey("topic")) {
+                        continue;
+                    }
+                    if (metricName.contains("byte-rate") && metric.tags().containsKey("node-id")) {
+                        continue;
+                    }
+                    double newValue = (Double)value.metricValue();
+                    if (metricName.equals("records-consumed-total")) {
+                        double prevValue = consumerMetrics.get(metricName);
+                        numberOfRecordsConsumed.increment(newValue - prevValue);
```

Comment:
Rather than tracking the previous value, do you think we could use the metric's built-in time feature?

https://kafka.apache.org/27/javadoc/org/apache/kafka/common/metrics/Measurable.html#measure-org.apache.kafka.common.metrics.MetricConfig-long-

Thus:

```
previousValue = ???.measure(???, lastTime);
```

It's not critical, but might be something to consider. Perhaps for a follow-on.

Reply:
Will do this later.